src/coord/timestamp.rs

```diff
@@ -408,10 +419,34 @@ impl Timestamper {
     fn create_rt_kinesis_connector(
         &self,
         _id: SourceInstanceId,
-        _kinc: KinesisSourceConnector,
+        kinc: KinesisSourceConnector,
     ) -> RtKinesisConnector {
-        error!("Timestamping is unsupported for kinesis sources");
-        RtKinesisConnector {}
+        let provider = StaticProvider::new(
+            kinc.access_key.clone(),
+            kinc.secret_access_key.clone(),
+            None,
+            None,
+        );
+
+        // todo@jldlaughlin: Use HttpClient::new_with_config() to support a TLS-enabled client
+        let request_dispatcher = HttpClient::new().unwrap();
+        let r: Region = kinc
+            .region
+            .parse()
+            .unwrap_or_else(|_| panic!("Failed to parse AWS region: {}", kinc.region));
+        let client = KinesisClient::new_with(request_dispatcher, provider, r);
+
+        // Each consumer name must be unique within a stream.
+        // todo@jldlaughlin: Add a random string at the end, too?
```

Comment:
Perhaps follow the kafka consumer format here, where you use a string like "materialize-rt-<SOURCE INSTANCE ID>"?

Reply:
I'm updating it to `materialize-rt-kinesis-`, in the future I'll grab stream info out of the ARN.